poc_iot_verifier/src/loader.rs

```diff
@@ -184,7 +185,7 @@ impl Loader {
             ),
         }
         tracing::info!("creating beacon xor filter");
-        let beacon_packet_data = xor_data.into_inner();
+        let beacon_packet_data: Vec<u64> = xor_data.into_inner().into_iter().unique().collect();
```

Comment:
This should be slightly faster, and doesn't require hashing like the itertools's version.

```suggestion
        let mut beacon_packet_data = xor_data.into_inner();
        beacon_packet_data.sort();
        beacon_packet_data.dedup();
```